### PR TITLE
Implement incremental packed list (basic type list/vector) hashing

### PR DESCRIPTION
**Pull request by Nashatyrev**
This is the completion of PR #115 
Also addresses one item of the issue #120 

No special tricks comparing to non-packed lists except the support for serializing subLists were added

### CODE REVIEW

File: ssz/src/main/java/org/ethereum/beacon/ssz/visitor/SSZIncrementalHasher.java [hunk 1]
```diff
@@ -88,19 +89,55 @@ private MerkleTrie updateNonPackedTrie(
...
+        idx -> childVisitor.apply(idx, type.getChild(value, idx)).getFinalRoot(),
+        type.getChildrenCount(value),
+        merkleTree,
+        elementsUpdated);
+  }
+
+  private MerkleTrie updatePackedTrie(
+      SSZListType type,
+      Object value,
+      BiFunction<Integer, Object, MerkleTrie> childVisitor,
```
Comment (mkalinin) [inline ssz/src/main/java/org/ethereum/beacon/ssz/visitor/SSZIncrementalHasher.java]: It appears that `childVisitor` is not used.

File: ssz/src/main/java/org/ethereum/beacon/ssz/visitor/SSZIncrementalHasher.java [hunk 2]
```diff
@@ -162,4 +190,20 @@ private MerkleTrie copyWithSize(MerkleTrie trie, int newChunksCount) {
       return new MerkleTrie(newNodes);
     }
   }
+
+  private BytesValue serializePackedChunk(SSZListType basicListType, Object listValue, int chunkIndex) {
+    int typeSize = basicListType.getElementType().getSize();
+    int valsPerChunk = bytesPerChunk / typeSize;
+    if (valsPerChunk * typeSize != bytesPerChunk) {
+      throw new UnsupportedOperationException("");
```
Comment (mkalinin) [inline ssz/src/main/java/org/ethereum/beacon/ssz/visitor/SSZIncrementalHasher.java]: Let's add some message here.

**Review by mkalinin — APPROVED** [review]
LGTM! There is a couple of cosmetic fixes might worth to be done.

**Review by zilm13 — APPROVED** [review]
LGTM

**Pull request merged**